Make awaiting `Join` trigger not return `Task` result (#3931)
cocotb/cocotb: docs/source/newsfragments/3931.removal.rst

> ``cocotb.triggers.Join.retval`` was removed. If you need to get the of the task you've joined you can use :meth:`Task.result() <cocotb.task.Task.result>`.

Reviewer: ```suggestion
``cocotb.triggers.Join.retval`` was removed. Use :attr:`Join.task <cocotb.triggers.Join.task>` to get the joined task and :meth:`Task.result() <cocotb.task.Task.result>` to get its result.
```
